### ISSUE TITLE
Bug in applyAction method of the Button component

### ISSUE THREAD

**Issue by danbuchal** [magento/magento2#7445]
### Preconditions
Currently using Magento 2.1.1

### Steps to reproduce
I observed this issue when attempting to open a modal with a link and then call a method that takes a parameter.

This can be reproduced with the built in 'setTitle' method.

The following is from the modifier for my UI Component:
``` php
return [
    'arguments' => [
        'data' => [
            'config' => [
                'title' => 'A Title!',
                'formElement' => Container::NAME,
                'componentType' => Container::NAME,
                'component' => 'Magento_Ui/js/form/components/button',
                'actions' => [
                    [
                        'targetName' => 'ns=' . static::FORM_NAME . ', index=' . static::MY_MODAL,
                        'actionName' => 'setTitle',
                        'params' => [
                            'A New Title!'
                        ]
                    ],
                    [
                        'targetName' => 'ns=' . static::FORM_NAME . ', index=' . static::MY_MODAL,
                        'actionName' => 'openModal'
                    ]
                ],
                'displayAsLink' => true,
            ],
        ],
    ],
];
```

### Expected result
I should be able to:
1. Click the link and have the modal open 
2. Close the modal
3. Click the link again and have it open again with the correct title.

### Actual result
After first clicking the link everything appears normal.
If I close the modal and click the link again the Title is now "setTitle" (i.e. the actionName)

### Cause
I traced this back to the following line of code:
https://github.com/magento/magento2/blob/develop/app/code/Magento/Ui/view/base/web/js/form/components/button.js#L71

The actionName is added to the params array, but not removed.
As a result on the next call it is added again, which causes the issue described,

### Solution
I was able to work around the issue by extending the Button component and removing it, like so:
``` js
define([
    'Magento_Ui/js/form/components/button'
], function (Button) {
    'use strict';

    return Button.extend({
        /**
         * Apply action on target component,
         * but previously create this component from template if it is not existed
         *
         * @param {Object} action - action configuration
         */
        applyAction: function (action) {
            this._super(action);

            var params = action.params || [],
                actionName = action.actionName;

            // The core 'applyAction' prepends the 'actionName' to the params array, but does not remove it.
            if (params[0] === actionName ) {
                params.shift();
            }
        }
    });
});
```
and finally changing the config to use this new component:
``` php
'component' => 'MyCompany_Ui/js/form/components/button',
```

**Comment by tkacheva** [comment]
Issues is fixed in develop brunch https://github.com/magento/magento2/commit/bdf2d49b5b79b1c2d2611b14f4178d6c6f6e78a4 
In you face it again, please reopen ticket

**Comment by danbuchal** [comment]
@tkacheva I'm not sure I see how the commit you reference fixes the issue.  Can you please explain?
The root of the issue is here:
https://github.com/magento/magento2/blob/develop/app/code/Magento/Ui/view/base/web/js/form/components/button.js#L71

The actionName is added to the array, but should be removed after target.apply is called.

**Comment by que-etc** [comment]
@danbuchal I presume that @tkacheva meant this commit: https://github.com/magento/magento2/commit/55f41385237db25f301af3ed67c2f3574be40bd8 and [this change](https://github.com/magento/magento2/blame/develop/app/code/Magento/Ui/view/base/web/js/form/components/button.js#L61) specifically. In develop branch the initial `params` array is no longer being modified as the `actionName` property is added to the array's copy.

**Comment by danbuchal** [comment]
Thanks @que-etc, that makes much more sense.  Looks good!

**Comment by Ctucker9233** [comment]
Fix not included in 2.1.10
